src/core_plugins/metrics/server/lib/vis_data/request_processors/table/pivot.js

```diff
@@ -0,0 +1,53 @@
+import { get, set, last } from 'lodash';
+
+import basicAggs from '../../../../../common/basic_aggs';
+import getBucketSize from '../../helpers/get_bucket_size';
+import getTimerange from '../../helpers/get_timerange';
+import getIntervalAndTimefield from '../../get_interval_and_timefield';
+import getBucketsPath from '../../helpers/get_buckets_path';
+import bucketTransform from '../../helpers/bucket_transform';
+
+export default function pivot(req, panel) {
+  return next => doc => {
+    const { sort } = req.payload.state;
+    if (panel.pivot_id) {
+      set(doc, 'aggs.pivot.terms.field', panel.pivot_id);
+      set(doc, 'aggs.pivot.terms.size', panel.pivot_rows);
+      if (sort) {
+        const series = panel.series.find(item => item.id === sort.column);
+        const { timeField, interval } = getIntervalAndTimefield(panel, series);
+        const { bucketSize } = getBucketSize(req, interval);
+        const { to }  = getTimerange(req);
+        const metric = series && last(series.metrics);
+        if (metric && metric.type === 'count') {
+          set(doc, 'aggs.pivot.terms.order', { _count: sort.order });
+        } else if (metric && basicAggs.includes(metric.type)) {
+          const sortAggKey = `${metric.id}-SORT`;
+          const fn = bucketTransform[metric.type];
+          const bucketPath = getBucketsPath(metric.id, series.metrics)
+            .replace(metric.id, `${sortAggKey} > SORT`);
+          set(doc, `aggs.pivot.terms.order`, { [bucketPath]: sort.order });
+          set(doc, `aggs.pivot.aggs`, {
+            [sortAggKey]: {
+              filter: {
+                range: {
+                  [timeField]: {
+                    gte: to.valueOf() - (bucketSize * 1500),
```

Comment:
Also here, why change the value?

Reply:
This is here for sorting, the group by terms works the same way (https://github.com/elastic/kibana/blob/master/src/core_plugins/metrics/server/lib/vis_data/request_processors/series/split_by_terms.js#L26-L44). You can't sort a terms aggregation on a metric that is a child of a date histogram so you have to create a metric outside the date histogram to sort on with a filter that approximates the last bucket. I originally took the bucket size and multiplied by 1000 milliseconds but would occasionally get strange results; using 1500 milliseconds made things more reliable. Also notice this is slightly different because it's adding the time to the `gte` side and not the `lte` side.